docs/modules/pipelines/pages/job-monitoring.adoc

```diff
@@ -0,0 +1,52 @@
+= Monitoring Jobs
+:description: To ensure the reliability of streaming jobs and their workloads, try implementing some basic monitoring. With Hazelcast, you can monitor the status of streaming jobs to quickly identify job failures for troubleshooting.
+
+{description}
+
+== Monitoring for Status Changes
+
+There are two ways to monitor for job status changes. You can either poll the `Job.getStatus()` method separately for each job or add a `JobStatusListener` to your jobs.
+
+== Using a JobStatusListener
+
+The `JobStatusListener` has several advantages over polling for the job status. 
+
+- You can write custom code for execution on status change. For example, triggering an alert or writing the changed status to a dashboard.
+
+- You can identify jobs in an error state. By default, xref:troubleshoot:error-handling.adoc#processing-guarantees[all streaming jobs are suspended on failure]. When a job moves to the suspended status, the listener is passed a suspension cause; either an error description or requested by a user. 
+
+- You can also work out which were jobs canceled by a user, by calling the `isUserCancelled();` method. This method returns `true` if the user stopped the job, even though the job status is `FAILED`. 
+
+NOTE: The reasons for job suspensions and failures are also displayed on the Jobs page in Management Center.
+
+=== Adding a JobStatusListener
+
+To register the listener for a job, add the following to your client code:
+
+```java
+myJob.addStatusListener(JobStatusListener)
+```
+The following example creates a new job, registers the `JobStatusListener`, and uses a lambda expression to print out status changes to the stdout console, including whether they were requested by a user.
+
+```java
+JetService jet1 = hz1.getJet();
+JobConfig jobConfig = new JobConfig();
+String jobName = "sampleJob";
+jobConfig.setName(jobName);
+Job job = jet1.newJob(p, jobConfig);
+
+job.addStatusListener(event ->
+        System.out.printf("Job status changed: %s -> %s. User requested? %b%n",
+                event.getPreviousStatus(),
+                event.getNewStatus(),
+                event.isUserRequested())
+);
+```
+
+=== Removing a JobStatusListener
+
+To de-register the listener from a job, add the following to your client code:
+
+```java
+myJob.removeStatusListener(JobStatusListener)
```

Comment:
```suggestion
myJob.removeStatusListener(statusListenerId)
```
This method takes the `UUID` returned by `addStatusListener()` as an argument rather than an instance of `JobStatusListener`. See https://docs.hazelcast.org/docs/latest-dev/javadoc/com/hazelcast/jet/Job.html#removeStatusListener-java.util.UUID-

Reply:
Updated as requested.